src/Acuminator/Acuminator.Utilities/DiagnosticSuppression/SuppressionManager.cs

```diff
@@ -73,15 +73,20 @@ public static void Init(ISuppressionFileSystemService fileSystemService, IEnumer
 
 			var suppressionFiles = additionalFiles.Where(f => SuppressionFile.IsSuppressionFile(f.Path));
 
-			if (Instance != null)
-			{
-				throw new InvalidOperationException($"{typeof(SuppressionManager).Name} has been already initialized");
-			}
-
-			Instance = new SuppressionManager(fileSystemService, suppressionFiles);
-		}
-
-		public static void SaveSuppressionBase()
+            if (Instance?._fileSystemService != null)
```

Comment:
It might be possible that between `if` and `lock` statements another thread will initialize `Instance` property, so we need to double-check if `Instance?._fileSystemService` is null inside the lock.

Reply:
@dropsonic it might be so, but the logic inside `if` is fired if `Instance?._fileSystemService` is not null

Reply:
Well, now I thnik that monitor surely prolong life of locked object and it won't be GCed inside lock. So it should be safe, however, the double lock technique is quite standard - I don't thnik that it will hurt to add it for extra safety.